packages/flutter/test/material/segmented_button_test.dart

```diff
@@ -1122,6 +1129,111 @@ void main() {
+              ),
+              segments: const <ButtonSegment<int>>[
+                ButtonSegment<int>(
+                  value: 0,
+                  label: Text('Add'),
+                  icon: Icon(Icons.add),
+                ),
+                ButtonSegment<int>(
+                  value: 1,
+                  label: Text('Subtract'),
+                  icon: Icon(Icons.remove),
+                ),
+              ],
+              showSelectedIcon: false,
+              onSelectionChanged: enabled ? (Set<int> selected) {} : null,
+              selected: const <int>{0},
+            ),
+          ),
+        ),
+      );
+    }
+
+    // Test enabled button.
+    await tester.pumpWidget(buildButton());
+    expect(tester.getSize(find.byIcon(Icons.add)), const Size(iconSize, iconSize));
+    expect(iconStyle(tester, Icons.add).color, iconColor);
+
+    // Test disabled button.
+    await tester.pumpWidget(buildButton(enabled: false));
+    expect(iconStyle(tester, Icons.add).color, disabledIconColor);
+  });
+
+  // This is a test regression for https://github.com/flutter/flutter/issues/154798.
+  test('SegmentedButton.styleFrom constructor', () {
+    const Color foregroundColor = Color(0xFF000022);
+    const Color backgroundColor = Color(0xFF000011);
+    const Color disabledForegroundColor = Color(0xFF000033);
+    const Color disabledBackgroundColor = Color(0xFF000044);
+    const Color shadowColor = Color(0xFF000066);
+    const Color surfaceTintColor = Color(0xFF000077);
+    const Color iconColor = Color(0xFF000088);
+    const double iconSize = 32.0;
+    const Color disabledIconColor = Color(0xFF000099);
+    const Color overlayColor = Color(0xFF000055);
+    const double elevation = 10.0;
+    const TextStyle textStyle = TextStyle(fontSize: 21.0);
+    const EdgeInsetsGeometry padding = EdgeInsets.all(16.0);
+    const Size minimumSize = Size(88.0, 36.0);
+    const Size fixedSize = Size(100.0, 50.0);
+    const Size maximumSize = Size(120.0, 48.0);
+    const BorderSide side = BorderSide(width: 2.0, color: Color(0xFF0000AA));
+    final OutlinedBorder shape = RoundedRectangleBorder(borderRadius: BorderRadius.circular(8.0));
+    const MouseCursor enabledMouseCursor = SystemMouseCursors.click;
+    const MouseCursor disabledMouseCursor = SystemMouseCursors.basic;
+    const VisualDensity visualDensity = VisualDensity.compact;
+    const MaterialTapTargetSize tapTargetSize = MaterialTapTargetSize.shrinkWrap;
+    const Duration animationDuration = Duration(milliseconds: 200);
+    const bool enableFeedback = true;
+    const AlignmentGeometry alignment = Alignment.center;
+    const InteractiveInkFeatureFactory splashFactory = InkRipple.splashFactory;
+
+    SegmentedButton.styleFrom(
+      backgroundColor: backgroundColor,
+      foregroundColor: foregroundColor,
+      disabledForegroundColor: disabledForegroundColor,
+      disabledBackgroundColor: disabledBackgroundColor,
```

Comment:
Definitely not a huge deal, but one way to make this more concise would be to do

```dart
    SegmentedButton.styleFrom(
      foregroundColor: const Color(0xFF000022),
      backgroundColor: const Color(0xFF000011),
      disabledForegroundColor: const Color(0xFF000033),
      disabledBackgroundColor: const Color(0xFF000044),
      // ...
    );
```

It could make it easier to prevent confusion caused by e.g. `foregroundColor` and `backgroundColor` being defined in the opposite order of where they're referenced.